frontend-web/webclient/app/UCloud/FilesApi.tsx

```diff
@@ -660,6 +677,81 @@ class FilesApi extends ResourceApi<UFile, ProductStorage, UFileSpecification,
     fileSelectorModalStyle = largeModalStyle;
 }
 
+function SensitivityDialog({file, invokeCommand, reload}: {file: UFile; invokeCommand: InvokeCommand; reload: () => void;}): JSX.Element {
+    // Note(Jonas): It should be initialized at this point, but let's make sure.
+    if (!sensitivityTemplateId) {
+        sensitivityTemplateId = findTemplateId(file, FileSensitivityNamespace, FileSensitivityVersion);
+    }
+
+    const originalSensitivity = findSensitivity(file);
+    const selection = React.useRef<HTMLSelectElement>(null);
+    const reason = React.useRef<HTMLTextAreaElement>(null);
+
+    const onUpdate = React.useCallback(async (e: React.SyntheticEvent) => {
+        e.preventDefault();
+        e.stopPropagation();
+        try {
+            const value = selection.current?.value;
+            const reasonText = reason.current?.value ?? "No reason provided."
+            if (!value) return;
+            if (value === SensitivityLevelMap.INHERIT) {
+                // Find latest that is active and remove that one. At most one will be active.
+                const entryToDelete = file.status.metadata?.metadata[sensitivityTemplateId][0];
```

Comment:
The active one will have `.status.type` should be either `approved` or `not_required`. The current code just picks the latest entry, if I remember correctly.